### PR TITLE
Issue 1424

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
PR to address issue #1424. Tested on laptop with gcc/5.4 Serial backend and profiling disabled. Tpetra Serial build with profiling disabled going locally now, no issues so far (taking awhile). Will run spot check tomorrow morning.

**Comment by ndellingwood** [comment]
@nmhamster I can reissue with that commit as a separate PR. It is not related, added it while getting things up and running with a Tpetra build which exposed the extent bug.

**Comment by ndellingwood** [comment]
Serial build of Tpetra with profiling disabled completed without errors, all unit tests pass. 
For added confidence checked that KokkosCore_config.h and TpetraCore_config.h do not have `KOKKOS_ENABLED_PROFILING` sneaking around,
and from CMakeCache.txt: 
```
//CamelCase Compatibility setting for KOKKOS_ENABLE_PROFILING
Kokkos_ENABLE_Profiling:BOOL=OFF
```

**Comment by mhoemmen** [comment]
@ndellingwood Oddly enough, `Teuchos_config.h` has `KOKKOS_ENABLE_PROFILING` redundantly defined; see https://github.com/trilinos/Trilinos/issues/2271 .  I will submit a PR to Trilinos very shortly that will remove this.  Thanks to https://github.com/vbrunini for pointing this out!

**Comment by mhoemmen** [comment]
I just submitted a Trilinos PR to fix the aforementioned Teuchos issue: https://github.com/trilinos/Trilinos/pull/2272

File: core/src/impl/Kokkos_Profiling_Interface.hpp [hunk 1]
```diff
@@ -45,19 +45,20 @@
...
-#if defined(KOKKOS_ENABLE_PROFILING)
-
+#include <cinttypes>
 #include <cstddef>
 #include <Kokkos_Core_fwd.hpp>
 #include <string>
-#include <cinttypes>
 
-#include <impl/Kokkos_Profiling_DeviceInfo.hpp>
 #include <dlfcn.h>
```
Comment (ndellingwood) [inline core/src/impl/Kokkos_Profiling_Interface.hpp]: @dsunder said this needs to be moved back within the KOKKOS_ENABLE_PROFILING guard - some machines will not compile with dynamic libraries

**Comment by ndellingwood** [comment]
test_all_sandia does not allow for --with-options to disable profiling, but I checked that existing behavior was not broken at least with my changes. Started the tests earlier before @dsunder comment about dlfcn.h lib which needs fixed, but posting results for reference. 

```
[ndellin@kokkos-dev TestAllSandia]$ ../../config/test_all_sandia --spot-check
Running on machine: sems
Repository Status:  0c370b6a00ecfa5665d40b8caca1b9eba0a23e20 DualView: Rename 'extents' to 'extent'


Going to test compilers:  gcc/5.3.0 gcc/6.1.0 intel/17.0.1 clang/3.9.0 cuda/8.0.44
Testing compiler gcc/5.3.0
Testing compiler gcc/6.1.0
  Starting job gcc-5.3.0-OpenMP-release
  Starting job gcc-5.3.0-OpenMP-hwloc-release
  Starting job gcc-6.1.0-Serial-release
  PASSED gcc-5.3.0-OpenMP-release
Testing compiler intel/17.0.1
  Starting job gcc-6.1.0-Serial-hwloc-release
  PASSED gcc-6.1.0-Serial-release
  Starting job intel-17.0.1-OpenMP-release
  PASSED gcc-5.3.0-OpenMP-hwloc-release
Testing compiler clang/3.9.0
  Starting job intel-17.0.1-OpenMP-hwloc-release
  PASSED gcc-6.1.0-Serial-hwloc-release
  Starting job clang-3.9.0-Pthread_Serial-release
  PASSED clang-3.9.0-Pthread_Serial-release
Testing compiler cuda/8.0.44
  Starting job clang-3.9.0-Pthread_Serial-hwloc-release
  PASSED intel-17.0.1-OpenMP-release
  PASSED intel-17.0.1-OpenMP-hwloc-release
  PASSED clang-3.9.0-Pthread_Serial-hwloc-release
  Starting job cuda-8.0.44-Cuda_OpenMP-release
  PASSED cuda-8.0.44-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-3.9.0-Pthread_Serial-hwloc-release build_time=189 run_time=89
clang-3.9.0-Pthread_Serial-release build_time=201 run_time=351
cuda-8.0.44-Cuda_OpenMP-release build_time=495 run_time=547
gcc-5.3.0-OpenMP-hwloc-release build_time=262 run_time=303
gcc-5.3.0-OpenMP-release build_time=258 run_time=279
gcc-6.1.0-Serial-hwloc-release build_time=195 run_time=127
gcc-6.1.0-Serial-release build_time=208 run_time=339
intel-17.0.1-OpenMP-hwloc-release build_time=559 run_time=295
intel-17.0.1-OpenMP-release build_time=567 run_time=297
```

**Comment by ndellingwood** [comment]
Closing this PR and replacing with #1429 to separate the dualview fix from the profiling issue addressed here; that PR also includes @dsunder suggestion.

**Pull request closed without merging**